### PR TITLE
Expand get_decision_problem_results with kwargs

### PR DESCRIPTION
**Pull request by tengis-nrl**
Tests passing
image

### CODE REVIEW

**Comment by tengis-nrl** [comment]
image

File: test/test_simulation_results.jl [hunk 1]
```diff
@@ -690,10 +689,26 @@ function test_simulation_results_from_file(path::AbstractString, c_sys5_hy_ed, c
...
 
+    results_ed = get_decision_problem_results(
+        results,
+        "ED";
+        populate_system = true,
+        populate_units = IS.UnitSystem.DEVICE_BASE,
+    )
+    @test !isnothing(PSI.get_system(results_ed))
+    @test PSY.get_units_base(get_system(results_ed)) == "DEVICE_BASE"
+
```
Comment (GabrielKS) [inline test/test_simulation_results.jl:711]: Can you also test that the errors work as expected when we can't find the system file and when we set `populate_units` without setting `populate_system`?

**Review by GabrielKS — CHANGES_REQUESTED** [review]

File: src/simulation/simulation_results.jl [hunk 2]
```diff
@@ -193,12 +193,43 @@ Base.length(res::SimulationResults) =
     mapreduce(length, +, values(res.decision_problem_results))
 get_exports_folder(x::SimulationResults) = joinpath(x.path, "exports")
 
-function get_decision_problem_results(results::SimulationResults, problem)
+"""
```
[the remark this answers is not in the document]
Comment (tengis-nrl) [reply src/simulation/simulation_results.jl:196]: Looks like I lost the description at some point. Will add it back.

File: src/simulation/simulation_results.jl [hunk 3]
```diff
@@ -193,12 +193,43 @@ Base.length(res::SimulationResults) =
...
 
-    return results.decision_problem_results[problem]
+    results = results.decision_problem_results[problem]
+
+    if populate_system
+        get_system!(results)
+        (populate_units !== nothing) &&
+            PSY.set_units_base_system!(PSI.get_system(results), populate_units)
+    else
+        (populate_units !== nothing) && (populate_units !== IS.UnitSystem.NATURAL_UNITS) &&
```
[the remark this answers is not in the document]
Comment (tengis-nrl) [reply src/simulation/simulation_results.jl:224]: Could you check this sentence? I don't get it. 

This logic will throw an error if the the user manually sets populate_units =IS.UnitSystem.DEVICE_BASE, but not if the user manually sets populate_units =IS.UnitSystem.DEVICE_BASE, which is inconsistent.

[the remark this answers is not in the document]
Comment (GabrielKS) [reply src/simulation/simulation_results.jl]: Yeah

File: test/test_simulation_results.jl (hunk 1 above)
Comment (jd-lara) [reply test/test_simulation_results.jl:711]: once this test is implemented we can merge

**Review by jd-lara — APPROVED** [review]

**Comment by tengis-nrl** [comment]
image
The test for testing "file not found" error passes

File: test/test_simulation_results.jl (hunk 1 above)
Comment (tengis-nrl) [reply test/test_simulation_results.jl:727]: Hi @GabrielKS, could you confirm or deny the implementation of the test?

**Pull request merged**